examples/Azure_IoT_Central_ESP32/AzureIoT.h

```diff
@@ -0,0 +1,809 @@
+ * @remark       The states informed are simplified for ease-of-use of this client, not reflecting
+ *               all the detailed internal states possible within the Azure IoT client.
+ * 
+ * @param[in]    azure_iot             A pointer to the instance of `azure_iot_t` previously initialized by the caller.
+ *
+ * @return       azure_iot_status_t    One of four possible states as defined by `azure_iot_status_t`.
+ *                                     Please see the `azure_iot_status_t` documentation above for details.
+ */
+azure_iot_status_t azure_iot_get_status(azure_iot_t* azure_iot);
+
+/*
+ * @brief        Causes the Azure IoT client to perform its tasks for connecting and working with Azure IoT services.
+ * @remark       This function must be called frequently enough for the Azure IoT client to work properly.
+ *               That frequency should be enough to respect the timeouts implemented by the Azure IoT services for
+ *               its TCP connection and MQTT-protocol traffic with the client application.
+ *               Calling it once within a main loop() function of most embedded implementations should be enough.
+ *               The recommended minimum frequency is no less than once every 100 milliseconds.
+ * 
+ * @param[in]    azure_iot             A pointer to the instance of `azure_iot_t` previously initialized by the caller.
+ *
+ * @return                             Nothing.
+ */
+void azure_iot_do_work(azure_iot_t* azure_iot);
+
+/*
+ * @brief        Sends a telemetry payload to the Azure IoT Hub.
+ * 
+ * @param[in]    azure_iot    A pointer to the instance of `azure_iot_t` previously initialized by the caller.
+ * @param[in]    message      An az_span instance containing the buffer and size of the actual message to be sent.
+ *
+ * @return       int          0 on success, or non-zero if any failure occurs.
+ */
+int azure_iot_send_telemetry(azure_iot_t* azure_iot, az_span message);
+
+/**
+ * @brief Checks whether `span` is equal AZ_SPAN_EMPTY.
+ *
+ * @param[in]    azure_iot                         The pointer to the azure_iot_t instance that holds the state of the Azure IoT client.
+ * @param[in]    message                           An `az_span` with the message with the reported properties update
+ *                                                 (a JSON document formatted according to the DTDL specification).
+ *                                                 `message` gets passed as-is to the MQTT client publish function as the payload, so if 
+ *                                                 your MQTT client expects a null-terminated string for payload, make sure `message` is
+ *                                                 a null-terminated string.
+ * @param[in]    request_id                        An unique identification number to correlate the response with when
+ *                                                 `on_properties_update_completed` (set in azure_iot_config_t) is invoked.
```

Comment:
nit: these two params are swapped in the API

Reply:
Fixed.